### PR TITLE
perf: removing unneccesary clones and muts

### PR DESCRIPTION
**Pull request by aphelionz**
This PR is part of reconciling our `interledger-rs` fork with the upstream. It contains the `clone()` and `mut` removal that @ljedrz performed.

Note: Clippy may fail and I or somebody will likely introduce another PR to fix those as well.

### CODE REVIEW

**Comment by aphelionz** [comment]
Relies on #659

File: crates/interledger-stream/src/packet.rs [hunk 1]
```diff
@@ -371,7 +371,7 @@ impl<'a> fmt::Debug for Frame<'a> {
 }
 
 /// The Stream Frame types [as defined in the RFC](https://interledger.org/rfcs/0029-stream/#53-frames)
-#[derive(Debug, PartialEq, Clone, Copy)]
+#[derive(Debug, PartialEq, Clone)]
```
Comment (ljedrz) [inline crates/interledger-stream/src/packet.rs:374]: `Copy` is preferable here

File: crates/interledger-http/src/server.rs [hunk 2]
```diff
@@ -65,12 +65,13 @@ async fn ilp_over_http<S, I>(
...
     body: Bytes,
     store: S,
-    mut incoming: I,
+    incoming: I,
 ) -> Result<impl warp::Reply, warp::Rejection>
 where
     S: HttpStore,
     I: IncomingService<S::Account> + Clone,
 {
+    let mut incoming = incoming.clone();
```
Comment (ljedrz) [inline crates/interledger-http/src/server.rs:74]: this is undesirable; the value is consumed, and thus can be `mut`

File: crates/interledger-stream/src/server.rs [hunk 3]
```diff
@@ -178,16 +179,13 @@ where
...
-                                amount,
-                                destination,
-                                timestamp: DateTime::<Utc>::from(SystemTime::now()).to_rfc3339(),
-                            })
-                    }
+                    Ok(ref _fulfill) => store.publish_payment_notification(PaymentNotification {
+                        to_username,
+                        from_username,
+                        amount,
+                        destination: destination.clone(),
```
Comment (ljedrz) [inline crates/interledger-stream/src/server.rs:186]: this `clone()` looks redundant

File: crates/interledger-stream/src/server.rs [hunk 4]
```diff
@@ -162,6 +162,7 @@ where
         let to_username = request.to.username().clone();
         let from_username = request.from.username().clone();
         let amount = request.prepare.amount();
+        let store = self.store.clone();
```
Comment (ljedrz) [inline crates/interledger-stream/src/server.rs:165]: this shouldn't be needed

Comment (ljedrz) [inline crates/interledger-stream/src/server.rs:165]: this seems unnecessary

**Comment by aphelionz** [comment]
Made this in error :facepalm: closing

**Pull request closed without merging**